### ISSUE TITLE
[BUG] Data Race CUDTSocket::removeFromGroup vs CUDT::processData

### ISSUE THREAD

**Issue by maxsharabayko** [Haivision/srt#2975]
```shell
WARNING: ThreadSanitizer: data race (pid=5444)
  Write of size 8 at 0x7ba80005a068 by thread T1 (mutexes: write M226, write M223):
    #0 srt::CUDTSocket::removeFromGroup(bool) /srt/srtcore/api.cpp:3484 (srt-xtransmit+0x297177)
    #1 srt::CUDTUnited::garbageCollect(void*) /srt/srtcore/api.cpp:3336 (srt-xtransmit+0x28f3eb)

  Previous read of size 8 at 0x7ba80005a068 by thread T4:
    #0 srt::CUDT::processData(srt::CUnit*) /srt/srtcore/core.cpp:10422 (srt-xtransmit+0x378165)
    #1 srt::CRcvQueue::worker_ProcessAddressedPacket(int, srt::CUnit*, srt::sockaddr_any const&) /srt/srtcore/queue.cpp:1475 (srt-xtransmit+0x3f01f6)
    #2 srt::CRcvQueue::worker(void*) /srt/srtcore/queue.cpp:1254 (srt-xtransmit+0x3ecfb7)
```

SRT v1.5.3+ (54c002f6ad9e3643d5d8117d32c5dada9944c151).

### To Reproduce

Build srt-xtransmit with Thread Sanitizer (`-DCMAKE_CXX_FLAGS='-fsanitize=thread' -DWITH_COMPILER_TYPE=clang`).
The error is reported on the receiver.

```shell
srt-xtransmit generate srt://172.27.16.247:4200 srt://172.27.16.247:5200 --enable-metrics --loglevel note --sendrate 5Mbps --num 2000 --reconnect --concurrent-streams 2 --maxconns 6 

srt-xtransmit receive srt://:4200 srt://:5200 --loglevel note --enable-metrics --reconnect --maxconns 6 --concurrent-streams 2
```

Epic #1610 – Core Synchronization Issues.

**Comment by ethouris** [comment]
This problem should be now addressed in #2893. The busy counter is added in the function that dispatches the packet to the given socket. This socket will not be deleted as long as processing isn't finished, but if the socket will be moved to trash in the meantime, any next dispatching will not be possible, so no more busy-acquisition will be done.

**Comment by maxsharabayko** [comment]
Can't reproduce any more even with the stated SRT commit ID.

72303d7934f9c6b1cbe23c438672f0eba0f318cb (Jun 12)   - not reproducible

54c002f6ad9e3643d5d8117d32c5dada9944c151 (Jul 9)   - not reproducible
